lib/open-amp/resource_table.h

```diff
@@ -17,8 +17,17 @@ extern "C" {
 #if (CONFIG_OPENAMP_RSC_TABLE_NUM_RPMSG_BUFF > 0)
 
 #define VDEV_ID                 0xFF
+#ifdef CONFIG_IPM_RX_CHANNEL_ID
+#define VRING0_ID CONFIG_IPM_RX_CHANNEL_ID
+#else
 #define VRING0_ID 0 /* (master to remote) fixed to 0 for Linux compatibility */
+#endif
```

Comment:
`CONFIG_IPM_RX_CHANNEL_ID `has a default value set to 0 

Having only following code
```
#define VRING0_ID CONFIG_IPM_RX_CHANNEL_ID
```
should work in all cases, no need to use `#ifdef CONFIG_IPM_RX_CHANNEL_ID` ( TBC by tests)